pyproject.toml

```diff
@@ -11,16 +11,21 @@ keywords = [
 license = "Apache-2.0"
 
 [tool.poetry.dependencies]
-python = "<3.12,>=3.7.1"
-singer-sdk = { version="^0.27.0" }
-fs-s3fs = { version = "^1.1.1", optional = true }
+python = "<3.12,>=3.8"
+singer-sdk = { version="^0.28.0" }
+s3fs = { version = "^2023.5.0", optional = true}
+fsspec = "^2023.5.0"
+ruff = "^0.0.270"
+black = "^23.3.0"
+fs-s3fs = { version = "^1.1.1", optional = true}
+
+[tool.poetry.extras]
+s3=["s3fs"]
+s3-batch=["fs-s3fs"]
```

Comment:
Kind of interesting to make two seperate extras here I didn't think about that. What do you think about just shoving them both under the s3 extra? That's what the "standard" is for other taps today so it'd make sense for folks, and I think most of the overhead from the dependency is boto3 (and downstream from boto3) so the addition of fs-s3fs probably isn't a huge deal.

Reply:
I've combined them into a single extra.